stix_shifter_modules/sysdig/stix_transmission/connector.py

```diff
@@ -0,0 +1,217 @@
+                                                      "prev_page_token": str(prev_page_token)}
+                    else:
+                        if not return_obj.get('error') and return_obj.get('success') is not False:
+                            return_obj['success'] = True
+                            return_obj['data'] = []
+                else:
+                    if response_code == 401:
+                        message = 'cannot verify credentials'
+                    else:
+                        response_dict = json.loads(response_wrapper.read())
+                        message = response_dict.get('message')
+                    return_obj = self.exception_response(response_wrapper.code, message)
+            else:
+                return_obj['success'] = True
+                return_obj['data'] = []
+
+        except InvalidMetadataException as ex:
+            response_dict['code'] = 100
+            response_dict['message'] = f'Invalid metadata: {str(ex)}'
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+
+        except Exception as ex:
+            if "timeout_error" in str(ex):
+                response_dict['code'] = 408
+            response_dict['message'] = str(ex)
+            self.logger.error('error when getting search results: %s', ex)
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'],
+                                      connector=self.connector)
+        return return_obj
+
+    @staticmethod
+    def update_event(data):
+        """
+        Set the finding_type as 'threat' for the event.
+        Replace invalid <NA> tag with empty string.
+        Parse the fd.name and set the fields for network-traffic in the event.
+        """
+        for event in data:
+            fields = event["content"]["fields"]
+            # replace unsupported field value <NA> with empty string
+            for f in fields:
+                if fields[f] == '<NA>':
+                    fields[f] = ''
+
+            # Setting the finding_type for the event as 'threat'
+            event["finding_type"] = "threat"
+            # Parsing network fields from event data
+            if fields.get("evt.type") and fields["evt.type"] == "connect":
+                if "->" in fields["fd.name"]:
+                    source, destination = fields["fd.name"].split('->')
+                    event["direction"] = "out"
+                    event["clientIpv4"], event["clientPort"] = source.split(":")
+                    event["serverIpv4"], event["serverPort"] = destination.split(":")
+                elif "<-" in fields["fd.name"]:
+                    destination, source = fields["fd.name"].split('<-')
+                    event["direction"] = "in"
+                    event["clientIpv4"], event["clientPort"] = source.split(":")
+                    event["serverIpv4"], event["serverPort"] = destination.split(":")
```

Comment:
The direction of the network traffic changes the source and destination. I see, you have set the same source and destination for both directions. are you sure sysdig automatically changes the source and destination? If not, you need to change specifically change it.

Reply:
As confirmed by Sysdig support, format is always same {source_ip}:{source_port}->{destination_ip}:{destination_port} in both cases when the connection is inbound or outbound.
(i.e) source and destination automatically changing the positions.